system/mp/test/mptest.cpp

```diff
@@ -742,78 +887,124 @@ int main(int argc, char* argv[])
         else
         {
             while (i+1 < argc && i-1 < 1000) {
-                PrintLog("MPTEST: adding node %d, port = <%s>", i-1, argv[i+1]);
+                PrintLog("MPTEST: adding node %u, port = <%s>", i-1, argv[i+1]);
                 nodes[i-1] = createINode(argv[i+1], my_port);
                 i++;
             }
         }
 
-        PrintLog("MPTEST: num_nodes = %d", i-1);
+        tot_ranks = i-1;
+
+        IGroup *group = createIGroup(tot_ranks, nodes);
```

Comment:
minor: this would leak if exits early [e.g. if (die) .. return ]
Probably not new, bu should use Owned where possible and similarly for the ICommunicaotr on line 946